a bit crude hack, true, but on OpenSolaris we don't have endian.h or
something

--- contrib/pgcrypto/px.h
+++ contrib/pgcrypto/px.h
@@ -39,7 +39,14 @@
 #endif
 
 #ifndef BYTE_ORDER
-#error BYTE_ORDER must be defined as LITTLE_ENDIAN or BIG_ENDIAN
+#define	LITTLE_ENDIAN	1234	/* least-significant byte first (vax, pc) */
+#define	BIG_ENDIAN	4321	/* most-significant byte first (IBM, net) */
+#if defined(i386) || defined(__i386) || defined(__amd64)
+#define	BYTE_ORDER	LITTLE_ENDIAN
+#endif
+#if defined(sparc) || defined(__sparc)
+#define	BYTE_ORDER	BIG_ENDIAN
+#endif
 #endif
 
 /* keep debug messages? */
